src/frontend/data/websocket.ts

```diff
@@ -0,0 +1,29 @@
+import RobustWebSocket from 'altamoon-robust-websocket';
+
+import { addResource } from 'data/stores/generics';
+import { UploadableObject, Video } from 'types/tracks';
+import { WS_ENPOINT } from 'settings';
+import { modelName } from 'types/models';
+
+import { appData } from './appData';
+
+type WSMessageType = {
+  resource: UploadableObject;
+  type: modelName;
+};
+
+let videoWebsocket: RobustWebSocket;
```

Comment:
IMO we can't have the case where we can't switch from a video to an other. Not in a LTI scenario.
I would like to manage the websocket outside react life cycle, this is why I made this implementation. Otherwise a hook exists if we want to add it to react https://www.npmjs.com/package/react-use-websocket but for now I don't see the interest to do that.